### DISCUSSION TITLE
Error on ConductorBaseWithActiveItem, evento OnActivated/OnDeactivated not working

### DISCUSSION THREAD

**Discussion by pedrollanes** [canton7/Stylet#184] [Q&A] [answered]
I'm trying to use Conductor<IScreen>.Collection.OneActive but I can see in the uploaded code that ChangeActiveItem on ConductorBaseWithActiveItem has this code:
```cs
  if (newItem != null)
            {
                this.EnsureItem(newItem);

                if (this.IsActive)
                    ScreenExtensions.TryActivate(newItem);
                else
                    ScreenExtensions.TryDeactivate(newItem);
            }
```

But this is wrong, it would be:
```cs
  if (newItem != null)
            {
                this.EnsureItem(newItem);

                if (**!**this.IsActive)
                    ScreenExtensions.TryActivate(newItem);
                else
                    ScreenExtensions.TryDeactivate(newItem);
            }
```

Because of this, when I use ActiveItem = SomeScreen, in this screen is not firing OnActivated item.
Are you going to fix this?
Thank you!!!

**Comment by pedrollanes** [comment, marked as answer]
Solved, the problem was that the conductor parent was not Active... :'(

**Comment by canton7** [comment]
The code is correct, as you discovered. It only activates the child if the parent is active. So no, I'm not going to fix it 🙂 

Glad you found the solution!
